queue_job/tests/common.py

```diff
@@ -251,11 +252,14 @@ def _perform_graph_jobs(self, jobs):
 
     def _add_job(self, *args, **kwargs):
         job = Job(*args, **kwargs)
-        self.enqueued_jobs.append(job)
-
-        patcher = mock.patch.object(job, "store")
-        self._store_patchers.append(patcher)
-        patcher.start()
+        if not job.identity_key or all(
+            j.identity_key != job.identity_key for j in self.enqueued_jobs
+        ):
+            self.enqueued_jobs.append(job)
+
+            patcher = mock.patch.object(job, "store")
+            self._store_patchers.append(patcher)
+            patcher.start()
```

Comment:
Suggested change
```python
    with mock.patch.object(job, "store") as patcher:
        self._store_patchers.append(patcher)
```

**Context Manager:** Used the `with` statement to create a context for the mock patcher. This ensures that the patcher's cleanup (`stop()`) is automatically handled even if an exception occurs.

By using the `with` statement, you ensure that the patcher is properly started and stopped, avoiding potential issues with resource leaks or incorrect mocking behavior.